Tests/Functional/Drivers/Queue/PhpAmqpLibDriverTest.php

```diff
@@ -0,0 +1,244 @@
+     *
+     * @group channel-exception
+     */
+    public function testDeclareChannelWithoutConnnection()
+    {
+        $this->expectException(AMQPConnectionClosedException::class);
+        $this->expectExceptionMessage('Broken pipe or closed connection');
+        $this->assertTrue($this->driver->isConnected());
+        $this->driver->declareChannel();
+        $this->driver->destroy($this->createConsumer()->getName());
+        $this->assertFalse($this->driver->isConnected());
+        $this->driver->declareChannel();
+    }
+
+    /**
+     * @group format
+     */
+    public function testFormat()
+    {
+        $format = 'text/plain';
+        $this->driver->setFormat($format);
+        $this->assertEquals($format, $this->driver->getFormat());
+    }
+
+    /**
+     * @group amqp-connection
+     * @throws \AMQPConnectionException
+     */
+    public function testConnectWithoutData()
+    {
+        $this->driver->declareChannel();
+        $consumer = $this->createConsumer();
+        $this->driver->destroy($consumer->getName());
+        $this->driver->configure('', '', '', '');
+        $this->driver->connect(true);
+        $this->assertTrue($this->driver->isConnected());
+        $this->assertInstanceOf(AMQPStreamConnection::class, $this->driver->declareChannel()->getConnection());
+    }
+
+    /**
+     * @dataProvider getMessages
+     * @group amqp-consume-callback
+     */
+    public function testConsumeWaitNoBlock(MessageInterface $msg)
+    {
+        $this->prepareToConsume($msg);
+        $amqpMessage = null;
+
+        $callback = function($message) use (&$amqpMessage){
+            $amqpMessage = $message;
+            $queueMessage = new QueueMessage();
+            $queueMessage->setMessageId($message->getDeliveryTag());
+            $this->driver->ack($queueMessage);
+        };
+
+        $this->driver->consume($this->consumer->getName(), $this->queueName, $callback);
+        $this->driver->isConsuming();
+        $this->driver->waitNoBlock();
+
+        $this->assertInstanceOf(AMQPMessage::class, $amqpMessage);
+        $this->assertEquals($amqpMessage->delivery_info['routing_key'], $this->queueName);
+        $this->assertInstanceOf(AsyncQueueConsumer::class, $this->consumer);
+        $this->assertEquals($amqpMessage->delivery_info['consumer_tag'], $this->consumer->getName());
+        $this->assertContains('QueueMessage', $amqpMessage->getBody());
+    }
+
+    /**
+     * Prepare the data and class to consume a message
+     *
+     * @param QueueMessage $msg
+     * @return mixed
+     */
+    public function prepareToConsume(MessageInterface $msg)
+    {
+        $this->endpoint = $this->createMock(EndpointInterface::class);
+        $msgIn = $this->createQueueMessage($msg);
+        $msgIn->addHeader('test_header', '12345');
+        $this->driver->send($msgIn);
+        $this->consumer = $this->createConsumer();
+        $this->driver->declareChannel();
```

Comment:
This method here is not needed because the `$this->driver->send()` already has inside the call to `declareChannel()`

Reply:
This flow is related to 2 flows that should run separately (send/consume). Because of this I've been including a disconnect and connect call to "restart" the connection and take a different channel to consum.

Reply:
I would like to discuss this personally

Reply:
Fixed.